analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.Implementation.cs

```diff
@@ -24,100 +24,76 @@
 
 namespace SonarAnalyzer.Helpers
 {
-    internal abstract partial class KnownType
+    internal sealed partial class KnownType
     {
-        public string TypeName { get; private init; }
+        private readonly IList<string> namespaceParts;
+        private readonly string[] genericParameters;
 
-        internal abstract bool Matches(ITypeSymbol symbol);
+        public string TypeName { get; }
+        public string FullName { get; }
+        public bool IsArray { get; init; }
 
-        internal sealed class RegularKnownType : KnownType
+        internal KnownType(string fullName, params string[] genericParameters)
```

Comment:
All these `internal` should be `public` - forward from https://github.com/SonarSource/sonar-dotnet/pull/5689#discussion_r887949530